lang/csharp/src/apache/main/Reflect/ClassCache.cs

```diff
@@ -251,11 +252,31 @@ public void LoadClassCache(Type objType, Schema s)
                     LoadClassCache(objType.GenericTypeArguments[1], ms.ValueSchema);
                     break;
                 case NamedSchema ns:
-                    EnumCache.AddEnumNameMapItem(ns, objType);
+                    // add the non-nullable type to the the enum map, not the nullable one
+                    if (objType.IsGenericType && objType.GetGenericTypeDefinition() == typeof(Nullable<>))
+                    {
+                        EnumCache.AddEnumNameMapItem(ns, objType.GenericTypeArguments[0]);
+                    }
+                    else
+                    {
+                        EnumCache.AddEnumNameMapItem(ns, objType);
+                    }
+
                     break;
                 case UnionSchema us:
-                    if (us.Schemas.Count == 2 && (us.Schemas[0].Tag == Schema.Type.Null || us.Schemas[1].Tag == Schema.Type.Null) && objType.IsClass)
+                    if (us.Schemas.Count == 2 &&
+                        (us.Schemas[0].Tag == Schema.Type.Null || us.Schemas[1].Tag == Schema.Type.Null) &&
+                        (objType.IsClass
+                         || (objType.IsGenericType
+                             && (
+                                 objType.GetGenericTypeDefinition() == typeof(Nullable<>)
+                                 || objType.GetGenericTypeDefinition() == typeof(IList<>)
```

Comment:
Isn't this a part of https://github.com/Snagajob/avro/pull/4 (which of course is fine for our copy, but may want to fix for the one you submitted)

Reply:
Not exactly.  It's related, yes.  This fixes the Reflect class to support nulalble lists (arrays).  While #4 enables the GenericReaderWriter (which processes GenericRecords) to also support IList - currently it only supports Array.